### ISSUE TITLE
Use moge

### ISSUE THREAD

**Issue by kexul** [jiah-cloud/Align3R#5]
Hi, is it possible to use moge https://github.com/microsoft/MoGe as the backend? Which seems to be less memory intensive than dust3r.
By the way, how much vram is needed using the default configuration? I remembered that monst3r is very memory intensive.

**Comment by kexul** [comment]
OOM during `init edge (48,49*) score=13.76556396484375` in my 4090 with 24G Vram.

**Comment by jiah-cloud** [comment]
Yes, for videos with too many frames (more than ~30), it may run out of memory (OOM). To address this, you can use the Hierarchical Optimization (the details can be found in our paper), which significantly reduces memory consumption. This optimization is already implemented in pose_test.sh and depth_test.sh. We will also implement this function in demo.sh recently.

**Comment by jiah-cloud** [comment]
Moge is a great work! We will consider incorporating moge into our model.

> Hi, is it possible to use moge https://github.com/microsoft/MoGe as the backend? Which seems to be less memory intensive than dust3r. By the way, how much vram is needed using the default configuration? I remembered that monst3r is very memory intensive.

**Comment by kexul** [comment]
> Yes, for videos with too many frames (more than ~30), it may run out of memory (OOM). To address this, you can use the Hierarchical Optimization (the details can be found in our paper), which significantly reduces memory consumption. This optimization is already implemented in pose_test.sh and depth_test.sh. We will also implement this function in demo.sh recently.

Thanks for your work! Hopefully it can be incorporated soon! I've tried delete some model and using gc.collect() in the process, but it did not help much.

**Comment by jiah-cloud** [comment]
> Yes, for videos with too many frames (more than ~30), it may run out of memory (OOM). To address this, you can use the Hierarchical Optimization (the details can be found in our paper), which significantly reduces memory consumption. This optimization is already implemented in pose_test.sh and depth_test.sh. We will also implement this function in demo.sh recently.



> > Yes, for videos with too many frames (more than ~30), it may run out of memory (OOM). To address this, you can use the Hierarchical Optimization (the details can be found in our paper), which significantly reduces memory consumption. This optimization is already implemented in pose_test.sh and depth_test.sh. We will also implement this function in demo.sh recently.
> 
> Thanks for your work! Hopefully it can be incorporated soon! I've tried delete some model and using gc.collect() in the process, but it did not help much.

We have implemented this function in `demo.sh`. You can try it by using the flag `--mode='eval_pose_h'`. Thank you!

**Comment by kexul** [comment]
Thanks!
